WordPress/Helpers/RulesetPropertyHelper.php

```diff
@@ -43,15 +43,25 @@ final class RulesetPropertyHelper {
 	 * @since 2.0.0  No longer supports custom array properties which were incorrectly
 	 *               passed as a string.
 	 * @since 3.0.0  Moved from the Sniff class to this class.
+	 * @since 3.0.2  Added a new parameter to lowercase array keys and values.
 	 *
-	 * @param array $custom Custom list as provided via a ruleset.
-	 * @param array $base   Optional. Base list. Defaults to an empty array.
-	 *                      Expects `value => true` format when `$flip` is true.
-	 * @param bool  $flip   Optional. Whether or not to flip the custom list.
-	 *                      Defaults to true.
+	 * @param array $custom        Custom list as provided via a ruleset.
+	 * @param array $base          Optional. Base list. Defaults to an empty array.
+	 *                             Expects `value => true` format when `$flip` is true.
+	 * @param bool  $flip          Optional. Whether or not to flip the custom list.
+	 * @param bool  $lowercasekeys Optional. Whether to lowercase keys and values in the resulting array.
+	 *                             Defaults to false.
```

Comment:
Can we avoid the addition of another boolean parameter?
- https://alexkondov.com/should-you-pass-boolean-to-functions/
- https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/

Consider making it a string, like `$key_value_transform = 'none'` as the default, but 'lowercase_keys_only' or `uppercase_keys_values` or similar strings can be passed to determine the transformation of keys, values, or both.

Reply:
I'm not sure this is necessary, as this is a binary operation - either we lowercase the keys or not, there is no third option to choose from that would necessitate a string parameter imo.

Having a boolean value, to me, lowers the code complexity.

Reply:
There's no third option now, but there might be in the future. 

Having descriptive string values makes the code easier to read than `true, true` in function calls, and then having to work out what they mean.

Reply:
> Having descriptive string values makes the code easier to read than true, true in function calls, and then having to work out what they mean.

Well, named parameters in function calls is the real solution to that. We just have to wait for a PHP 8.0 minimum before we can use that...

I dislike using text labels for something which is a boolean thing. Too easy to have typos.
Which, of course, could be bypassed by using an enum for the values, but that will have to wait till PHP 8.1.

The PHP 5.4 compatible version of all that would be to have (class) constants which contain the valid values, be it boolean or string values, and pass those around instead.

I have no objection to a solution using class constants for the value of this parameter to make it more descriptive, but wonder if it's really necessary.